geode-cq/src/main/java/org/apache/geode/cache/query/cq/internal/ServerCQImpl.java

```diff
@@ -367,10 +367,12 @@ public void close(boolean sendRequestToServer) throws CqClosedException, CqExcep
       this.removeFromCqMap();
 
       // Stat update.
-      if (stateBeforeClosing == CqStateImpl.RUNNING) {
-        cqService.stats().decCqsActive();
-      } else if (stateBeforeClosing == CqStateImpl.STOPPED) {
-        cqService.stats().decCqsStopped();
+      if (!cqName.equals(serverCqName)) {
```

Comment:
How is this change addressing the issue. The CqName and serverCqName is not same on the server side CQ representation. As this class represent server side, when the CQ execution (close) comes here the cqName should not be the same, isn't the case, here...Please help us to understand the context/stack where you are seeing this check is needed.

Reply:
I need somehow to makes difference between servers, one on which cq is register and others on which it will process registerCq. The server on which cq is registered doing also incrementing active activeCqCount, but while closing or stopping it decrements on all, but with this check it will decrement just on the one server where it is incremented. I'm not sure if this is a good check but it works, and I asked on dev list for a help to differentiate on which decrement is needed, but It is concluded that this is a bug but without any idea how to differentiate it.

Reply:
Hi @agingade , any comments or other ideas to resolve this? :-)

Reply:
Again, sorry for delayed response. If needed you can always send a direct email to me.